kgraphql/src/main/kotlin/com/apurebase/kgraphql/schema/introspection/__Type.kt

```diff
@@ -26,10 +26,28 @@ interface __Type {
 
     // NON_NULL and LIST only
     val ofType: __Type?
-}
 
-fun __Type.typeReference(): String = when (kind) {
-    TypeKind.NON_NULL -> "${ofType?.typeReference()}!"
-    TypeKind.LIST -> "[${ofType?.typeReference()}]"
-    else -> name ?: ""
+    operator fun get(name: String): __Field? = null
+
+    fun isList(): Boolean = when {
+        kind == TypeKind.LIST -> true
+        ofType == null -> false
+        else -> (ofType as __Type).isList()
```

Comment:
Would `(ofType as? __Type)?.isList() ?: false` also work here without the second when case plus it would be a safe cast?

Reply:
Would work, yes. The current cast should also be safe, though, shouldn't it? I'll have a look in my next PR.